src/language/standaloneLanguageServerInstaller.ts

```diff
@@ -0,0 +1,219 @@
+import { ExtensionContext, FileSystemError, OutputChannel, Uri, window, workspace } from 'vscode';
+import { LanguageServerConstants } from '../constants';
+import * as os from 'os';
+import fetch from 'cross-fetch';
+import { SourceInstaller } from './sourceInstaller';
+import extract = require('extract-zip');
+import * as fs from 'fs';
+import got from 'got/dist/source';
+import { promisify } from 'util';
+import { Utils } from 'vscode-uri';
+const mkdirAsync = promisify(fs.mkdir);
+import { execFile } from 'child_process';
+import { Executable } from 'vscode-languageclient/node';
+import { getDotnetExecutablePath } from '../dotnet';
+import { Messages } from '../ui/messages';
+import path = require('path');
+const execFileAsync = promisify(execFile);
+const ArchiveFileName = 'dafny.zip';
+
+function getDafnyPlatformSuffix(): string {
+  switch(os.type()) {
+  case 'Windows_NT':
+    return 'win';
+  case 'Darwin':
+    return 'osx-10.14.2';
+  default:
+    return 'ubuntu-16.04';
+  }
+}
+
+export class StandaloneLanguageServerInstaller {
+  public constructor(
+    public readonly context: ExtensionContext,
+    private readonly preferredVersion: string,
+    public readonly statusOutput: OutputChannel
+  ) {}
+
+  public async getExecutable(args: string[]): Promise<Executable> {
+    const { path: dotnetExecutable } = await getDotnetExecutablePath();
+    let dllPath = LanguageServerConstants.GetDefaultPath(await this.getConfiguredVersion());
+    if(dllPath && !path.isAbsolute(dllPath)) {
+      dllPath = path.join(this.context.extensionPath, dllPath);
+    }
+
+    if(!fs.existsSync(dllPath)) {
+      const installed = await this.install();
+      if(!installed) {
+        throw new Error(Messages.Installation.Error);
+      }
+    }
+
+    return { command: dotnetExecutable, args: [ dllPath, ...args ] };
+  }
+
+  private async install(): Promise<boolean> {
```

Comment:
How did you ensure this works the same way? Can you at least give it a try?

Reply:
I will. It's in the testing list although still has a TODO.